src/PolynomialZonotope.jl

```diff
@@ -0,0 +1,228 @@
+
+"""
+    dim(pz::PolynomialZonotope)::Int
+
+Return the ambient dimension of a polynomial zonotope.
+
+### Input
+
+- `pz` -- polynomial zonotope
+
+### Output
+
+An integer representing the ambient dimension of the polynomial zonotope.
+"""
+dim(pz::PolynomialZonotope)::Int = length(pz.c)
+
+
+"""
+    σ(d::AbstractVector{N}, pz::PolynomialZonotope{N})::AbstractVector{N} where {N}
+
+Return the support vector of a polynomial zonotope along direction `d`.
+
+### Input
+
+- `d`  -- direction
+- `pz` -- polynomial zonotope
+
+### Output
+
+Vector representing the support vector.
+"""
+function σ(d::AbstractVector{N}, pz::PolynomialZonotope{N})::AbstractVector{N} where {N}
+    error("this function is not yet implemented")
+end
+
+"""
+    polynomial_order(pz::PolynomialZonotope)::Int
+
+Polynomial order of a polynomial zonotope.
+
+### Input
+
+- `pz` -- polynomial zonotope
+
+## Output
+
+The polynomial order, defined as the maximal power of the scalar factors ``β_i``.
+Usually denoted ``η``.
+"""
+polynomial_order(pz::PolynomialZonotope)::Int = length(pz.E)
+
+"""
+    order(pz::PolynomialZonotope)::Rational{Int}
+
+Order of a polynomial zonotope.
+
+### Input
+
+- `pz` -- polynomial zonotope
+
+## Output
+
+The order, defined as the number of generators divided by the ambient dimension.
+"""
+function order(pz::PolynomialZonotope)::Rational{Int}
+    η = polynomial_order(pz)  # number of generators
+    p = size(pz.E[1], 2)  # number of dependent factors
+    q = size(pz.G, 2)  # number of independent factors
+    ξ = sum(i -> binomial(p+i-1, i), 1:η) + q
+    n = dim(pz)
+    return ξ//n
+end
+
+"""
+    linear_map(M::Matrix, pz::PolynomialZonotope)
+
+Return the linear map of a polynomial zonotope.
+
+### Input
+
+- `M`  -- matrix
+- `pz` -- polynomial zonotope
+
+## Output
+
+Polynomial zonotope such that its center and generators are those of `pz`
+multiplied by the matrix `M`.
+"""
+function linear_map(M::Matrix, pz::PolynomialZonotope)
+    c = M * pz.c
+    E = [M*Ei for Ei in pz.E]
+    F = [M*Fi for Fi in pz.F]
+    G = M * pz.G
+    return PolynomialZonotope(c, E, F, G)
+end
+
+"""
+    scale(α::Number, pz::PolynomialZonotope)
+
+Return a polynomial zonotope modified by a scalar factor.
+
+### Input
+
+- `α` -- polynomial zonotope
+- `pz` -- polynomial zonotope
+
+## Output
+
+Polynomial zonotope such that its center and generators are multiples of those
+of `pz` by a factor ``α``.
+"""
+function scale(α::Number, pz::PolynomialZonotope)
+    c = α * pz.c
+    E = [α*Ei for Ei in pz.E]
+    F = [α*Fi for Fi in pz.F]
+    G = α * pz.G
+    return PolynomialZonotope(c, E, F, G)
+end
+
+"""
+    minkowski_sum(pz::PolynomialZonotope, z::Zonotope)
+
+Return the Minkowski sum of a polynomial zonotope and a usual zonotope.
+
+### Input
+
+- `pz` -- polynomial zonotope
+- `z`  -- usual zonotope
+
+## Output
+
+Polynomial zonotope.
+"""
+function minkowski_sum(pz::PolynomialZonotope, z::Zonotope)
```

Comment:
Add the symmetric method? It can fall back to this one.

Reply:
good. i imagine some macro for lazy people (like us) used as:

```julia
@symmetric function minkowski_sum(pz::PolynomialZonotope, z::Zonotope)
    c = pz.c + z.center
    G = [G z.generators]
    return PolynomialZonotope(c, pz.E, pz.F, G)
end
```

and builds the new symmetric function using this one.